libcaliptra/inc/caliptra_types.h

```diff
@@ -380,12 +390,14 @@ struct caliptra_invoke_dpe_resp {
     struct caliptra_resp_header cpl;
     uint32_t data_size;
     union {
+        struct dpe_resp_hdr                         resp_hdr;
         struct dpe_get_profile_response             get_profile_resp;
         struct dpe_initialize_context_response      initialize_context_resp;
         struct dpe_derive_context_response          derive_context_resp;
         struct dpe_certify_key_response             certify_key_resp;
         struct dpe_sign_response                    sign_resp;
         struct dpe_rotate_context_handle_response   rotate_context_handle_resp;
+        struct dpe_destroy_context_response         destroy_context_resp;
         struct dpe_get_certificate_chain_response   get_certificate_chain_resp;
         uint8_t                                     data[sizeof(struct dpe_certify_key_response)];
```

Comment:
In fact, sizeof(struct dpe_get_certificate_chain_response) is the longest.
I suggest two separate macros for Cmd and Resp:

#define DPE_RESP_DATA_MAX_SIZE                      2176
#define DPE_REQ_DATA_MAX_SIZE                       512

Reply:
Maybe I am missing something. It looks to me like dpe_certify_key_response is still the longest as it contains the certificate of 2048 bytes as well as a few additional fields over dpe_get_certificate_chain_response.

But won't this work in C? The union should still be the size of the largest element which is the goal here.
![image](https://github.com/user-attachments/assets/1170caf4-cb3c-4921-9f2b-844fcce43a7b)

Reply:
Yes, you're right. Certify_key_response is the longest.

In the DPE client side, we use "DPE_RESP_DATA_MAX_SIZE" in a compile-time check to be ensure that size of union is matching to what it is supposed to be, regardless of which response is bigger.